add bottom, top and search bar + new application icon

YannisWinkler opened the pull request.

YannisWinkler committed:
add bottom bar, top bar and change icon

YannisWinkler committed:
add search bar but not functional

YannisWinkler committed:
add end of UI search, bottom and top bar

MisterPeModder commented on app/src/main/java/eu/epitech/reyditech/Reyditech.kt:
```diff
@@ -1,11 +1,40 @@
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import java.time.format.TextStyle
```
Mauvais import

MisterPeModder commented on app/src/main/java/eu/epitech/reyditech/Reyditech.kt:
```diff
@@ -14,10 +43,124 @@ internal const val USER_AGENT: String = "android:eu.epitech.reyditech:v1.0.0 (by
+        },
+
+        content = {
+        },
```
Tu devrais mettre le contenu de l'application ici et non pas en bas de la fonction, c'est probablement pour ca que les posts passaient au dessus des barres

YannisWinkler committed:
format files and fix little mistakes

MisterPeModder commented on app/src/main/java/eu/epitech/reyditech/Reyditech.kt:
```diff
@@ -14,10 +43,124 @@ internal const val USER_AGENT: String = "android:eu.epitech.reyditech:v1.0.0 (by
+                        )
+                    }
+                    Button(
+                        onClick = { println("Profil") /*navController.navigate("hot")*/ },
```
N'utilise pas println, mais Log.i()
De plus c'est `Profile` en anglais

MisterPeModder commented on app/src/main/java/eu/epitech/reyditech/Reyditech.kt:
```diff
@@ -14,10 +43,124 @@ internal const val USER_AGENT: String = "android:eu.epitech.reyditech:v1.0.0 (by
+                        )
+                    }
+                    Button(
+                        onClick = { println("Profil") /* navController.navigate("hot")*/ },
```
Même comm qu'au dessus

MisterPeModder commented on app/src/main/java/eu/epitech/reyditech/screens/MainScreen.kt:
```diff
@@ -63,7 +60,7 @@ private fun MainScreenUI(
 ) {
     Theme {
-        Box(contentAlignment = Alignment.TopCenter, modifier = Modifier.fillMaxSize()) {
+        Box(contentAlignment = Alignment.TopCenter, modifier = Modifier.fillMaxSize().padding(bottom = 60.dp, top = 65.dp)) {
```
J'ai pas lancé sur mon tel, mais quelque chose me dit que tu viens de hardcoder les dimensions de ton tel

MisterPeModder commented on app/src/main/java/eu/epitech/reyditech/screens/MainScreen.kt:
```diff
@@ -63,7 +60,12 @@ private fun MainScreenUI(
+            contentAlignment = Alignment.TopCenter,
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(bottom = 60.dp, top = 65.dp)
```
Mon petit doigt me dit que tu à hardcodé les dimensions de ton tel, regarde le commentaire sur le Scafold de Reyditech()

YannisWinkler replied:
c'est une possibilité envisageable

MisterPeModder commented on app/src/main/java/eu/epitech/reyditech/Reyditech.kt:
```diff
@@ -14,10 +31,114 @@ internal const val USER_AGENT: String = "android:eu.epitech.reyditech:v1.0.0 (by
+            })
+    },
+
+        content = {}, bottomBar = {
```
Tu devrais mettre le contenu de l'application dans `content` plutot qu'en bas de la fonction, c'est probablement pour ca que les posts passaient au dessus des barres du haut et du bas.

MisterPeModder commented on app/src/main/java/eu/epitech/reyditech/Reyditech.kt:
```diff
@@ -14,10 +31,114 @@ internal const val USER_AGENT: String = "android:eu.epitech.reyditech:v1.0.0 (by
+                }
+
+            }
+        })
```
Formatte ton fichier!

MisterPeModder replied:
`CTRL+ALT+L` ***x2***

MisterPeModder reviewed (changes requested):
Vois les commentaires que j'ai laissé

YannisWinkler committed:
remove hardcoding mainscreen

MisterPeModder committed:
Add formatting configuration & format Reyditech.kt

YannisWinkler committed:
new files

MisterPeModder commented on app/src/main/java/eu/epitech/reyditech/Reyditech.kt:
```diff
@@ -14,16 +33,121 @@ internal const val USER_AGENT: String = "android:eu.epitech.reyditech:v1.0.0 (by
+                                println(searchparam)
+                                searchparam = it
+                            },
+                            label = { Text(text = "search") },
```
Le text est pas traduit
Pro tip: met ton cursor sur la string, fait ALT+ENTER, puis clique sur extract to resource

MisterPeModder commented on app/src/main/java/eu/epitech/reyditech/Reyditech.kt:
```diff
@@ -14,16 +33,121 @@ internal const val USER_AGENT: String = "android:eu.epitech.reyditech:v1.0.0 (by
 internal fun Reyditech() {
     val navController = rememberNavController()
+    var searchparam by remember { mutableStateOf(TextFieldValue("")) }
+    Theme {
```
J'ai mis une instance de Theme() ici pour l'instant, mais pour ta culture, c'est tres fortement déconseille par la doc d'Android.
Le composant a la racine de toute l'application doit être NavHost, pas Scafold (parce que il peut avoir des Screens sans Scafold, genre la page de login).
Laisse ca comme ca pour l'instant je m'en occupe dans une autre PR.

YannisWinkler replied:
vasy je m'en souviendrai, mais j'avais trouvé ça dans la doc de jetpack compose

YannisWinkler committed:
string

MisterPeModder committed:
Extract harcoded strings to resources

MisterPeModder committed:
Refactor Reyditech() to make NavHost the root component again

The application scaffold should be handled by the individual screens by using the new ReyditechScaffold() component.
